### PR TITLE
linux_4_19: drop

### PR DESCRIPTION
**Pull request by fabianhjr**
## Description of changes

Documented here as something to do alongside the feature freeze announcement about 8 weeks before scheduled release of next stable: https://nixos.github.io/release-wiki/Feature-Freeze-Announcement.html?highlight=kernel#pre-release-cleanupza

> Remove attributes that we know we will not be able to support, especially if there is a stable alternative. E.g. Check that our Linux kernels’ [projected end-of-life](https://www.kernel.org/category/releases.html) are after our release projected end-of-life.

> The feature freeze announcement should be done 8 weeks before the release to inform subsystem maintainers about the upcoming feature freeze

We are already about 8 weeks to release
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by K900** [comment]
Is it over?

**Comment by fabianhjr** [comment]
```
4.19 	Greg Kroah-Hartman & Sasha Levin 	2018-10-22 	Dec, 2024
```

As per release procedures for stable LTS kernels that reach EOL during release must be dropped before branch off. (though probably can wait a bet closer to said branch off)

**Comment by fabianhjr** [comment]
Documented here as something to do after the feature freeze announcement: https://nixos.github.io/release-wiki/Feature-Freeze-Announcement.html?highlight=kernel#pre-release-cleanup

> Remove attributes that we know we will not be able to support, especially if there is a stable alternative. E.g. Check that our Linux kernels’ [projected end-of-life](https://www.kernel.org/category/releases.html) are after our release projected end-of-life.

> The feature freeze announcement should be done 8 weeks before the release to inform subsystem maintainers about the upcoming feature freeze

We are already about 8 weeks to release

**Comment by RossComputerGuy** [comment]
Result of `nixpkgs-review pr 343283` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.pktgen</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_5_10.oci-seccomp-bpf-hook</li>
    <li>linuxKernel.packages.linux_5_10.oci-seccomp-bpf-hook.man</li>
    <li>linuxKernel.packages.linux_5_10.odp-dpdk</li>
  </ul>
</details>

**Comment by RossComputerGuy** [comment]
`inuxKernel.packages.linux_5_10.pktgen` fails due to:
```
pktgen> FAILED: app/pktgen.p/pktgen-arp.c.o
pktgen> gcc -Iapp/pktgen.p -Iapp -I../app -Ilib/common -I../lib/common -Ilib/utils -I../lib/utils -Ilib/vec -I../lib/vec -Ilib/plugin -I../lib/plugin -Ilib/cli -I../lib/cli -Ilib/lua -I../lib/lua -I/nix/store/kva1f093lwlncwl2cvvj7b6cln3qv7cs-dpdk-23.11/include -I/nix/store/hy6ywqyzl6biizivil31isdpb5xrv12v-libbsd-0.12.2-dev/include -I/nix/store/k1laas1kx62kdmlnhnmqmandb8i0wiin-numactl-2.0.18-dev/include -I/nix/store/z50i62qdg6d8l234n1pfsbajq6ig3jih-libpcap-1.10.4/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -Werror -DALLOW_EXPERIMENTAL_API -D_GNU_SOURCE -Wno-pedantic -Wno-format-truncation -pthread -include rte_config.h -march=armv8-a+crc -moutline-atomics '-D__PROJECT_VERSION="23.10.0"' -MD -MQ app/pktgen.p/pktgen-arp.c.o -MF app/pktgen.p/pktgen-arp.c.o.d -o app/pktgen.p/pktgen-arp.c.o -c ../app/pktgen-arp.c
pktgen> In file included from /nix/store/970mrbrvg7gnbhgm59smc2vmc5wv91s3-glibc-2.39-52-dev/include/string.h:548,
pktgen>                  from ../lib/cli/cli_scrn.h:22,
pktgen>                  from ../app/pktgen-arp.c:9:
pktgen> In function 'memcpy',
pktgen>     inlined from 'pktgen_process_arp' at ../app/pktgen-arp.c:146:17:
pktgen> /nix/store/970mrbrvg7gnbhgm59smc2vmc5wv91s3-glibc-2.39-52-dev/include/bits/string_fortified.h:29:10: error: argument 1 null where non-null expected [-Werror=nonnull]
pktgen>    29 |   return __builtin___memcpy_chk (__dest, __src, __len,
pktgen>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
pktgen>    30 |                                  __glibc_objsize0 (__dest));
pktgen>       |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
pktgen> /nix/store/970mrbrvg7gnbhgm59smc2vmc5wv91s3-glibc-2.39-52-dev/include/bits/string_fortified.h:29:10: note: in a call to built-in function '__builtin_memcpy'
pktgen> cc1: all warnings being treated as errors
```

**Comment by fabianhjr** [comment]
Will investigatte, thought this would be 0 rebuilds :eyes:

**Comment by fabianhjr** [comment]
Now I am confused why nixpkgs-review attempted 4 builds and ofborg detects this as 0 rebuilds.

rereading the changes seems like ofborg is correct but weird nonetheless.

**Comment by RossComputerGuy** [comment]
> Now I am confused why nixpkgs-review attempted 4 builds and ofborg detects this as 0 rebuilds.

Likely something with aliases?

**Comment by fabianhjr** [comment]
Looking and past drops this is a bit early by about 1-2 weeks however I don't think there is much expectation to retain this LTS much longer nor until branchoff.

Any opinions on whether to hold off or go ahead?

**Comment by RossComputerGuy** [comment]
I have no objections to merging it now, we don't have to always conform to the timeline. If someone wants to merge this, they're free to do so.

File: pkgs/top-level/linux-kernels.nix [hunk 1]
```diff
@@ -297,9 +279,10 @@ in {
 
     linux_xanmod_tt = throw "linux_xanmod_tt was removed because upstream no longer offers this option";
 
-    linux_5_18_hardened = throw "linux 5.18 was removed because it has reached its end of life upstream";
-    linux_5_19_hardened = throw "linux 5.19 was removed because it has reached its end of life upstream";
-    linux_6_0_hardened = throw "linux 6.0 was removed because it has reached its end of life upstream";
+    linux_4_14_hardened = throw "linux 4.14 was removed because it will reach its end of life within 23.11";
+    linux_4_19_hardened = throw "linux 4.19 was removed because it will reach its end of life within 24.11";
+    linux_6_4_hardened = throw "linux 6.4 was removed because it has reached its end of life upstream";
+    linux_6_5_hardened = throw "linux 6.5 was removed because it has reached its end of life upstream";
```
Comment (Ma27) [inline pkgs/top-level/linux-kernels.nix:285]: Did we originally forget about the linux_6_5_hardened alias?

**Review by Ma27 — CHANGES_REQUESTED** [review]
LGTM, thank you.

This needs a note in the release notes though.

For the future, it would be a little easier to review if the removal of old aliases would be in a second commit, but I won't block on that now since I'm done reviewing :)

**Comment by Ma27** [comment]
> about 1-2 weeks however I don't think there is much expectation to retain this LTS much longer nor until branchoff.

I'd be a little surprised if people are on unstable _and_ use such an old kernel. I think this is mostly relevant for people following stable. So fine by me.

File: pkgs/top-level/linux-kernels.nix (hunk 1 above)
Comment (fabianhjr) [reply pkgs/top-level/linux-kernels.nix:285]: aparently, normalised between hardened and upstream aliases but also hardened hasn't been updated in a while

**Comment by fabianhjr** [comment]
Split the change over 3 commits: drop, cleanup aliases, and cleanup common-config

**Comment by fabianhjr** [comment]
Looking at the previous three release notes there is no mention of dropped kernel versions.

I could only find the mention of the drop of 4.9 on 22.11

Feel like it is more important to mention when the default changes than each version dropped since the throw 

<details><summary>Search</summary>
<p>

```
rg -i "(linux|kernel)" nixos/doc/manual/release-notes/rl-2405.section.md
11:- The default kernel package has been updated from 6.1 to 6.6. All supported kernels remain available.
13:- For each supporting version of the Linux kernel, firmware blobs
45:- Cinnamon has been updated to 6.0. Please be aware that the [Wayland session](https://blog.linuxmint.com/?p=4591) is still experimental in this release and could potentially [affect Xorg sessions](https://blog.linuxmint.com/?p=4639). We suggest a reboot when switching between sessions.
80:- [clatd](https://github.com/toreanderson/clatd), a CLAT / SIIT-DC Edge Relay implementation for Linux.
174:- [ryzen-monitor-ng](https://github.com/mann1x/ryzen_monitor_ng), a desktop AMD CPU power monitor and controller, similar to Ryzen Master but for Linux. Available as [programs.ryzen-monitor-ng](#opt-programs.ryzen-monitor-ng.enable).
176:- [ryzen-smu](https://gitlab.com/leogx9r/ryzen_smu), Linux kernel driver to expose the SMU (System Management Unit) for certain AMD Ryzen Processors. Includes the userspace program `monitor_cpu`. Available at [hardward.cpu.amd.ryzen-smu](#opt-hardware.cpu.amd.ryzen-smu.enable).
340:- `inetutils` now has a lower priority to avoid shadowing the commonly-used `util-linux`. If one wishes to restore the default priority, simply use `lib.setPrio 5 inetutils` or override with `meta.priority = 5`.
362:- `linuxPackages_testing_bcachefs` is now fully deprecated by `linuxPackages_latest`, and is therefore no longer available.
545:- `services.redis.vmOverCommit` now defaults to `true` and no longer enforces Transparent Hugepages (THP) to be disabled. Redis only works with THP configured to `madvise` which is the kernel's default.
604:   Support for these features was added in kernel 4.9 and 5.19 and nrext64 was deemed stable in kernel 6.5.
605:   Format your filesystems with `mkfs.xfs -i nrext64=0`, if they need to be readable by GRUB2 before 2.12 or kernels older than 5.19.
642:- `boot.kernel.sysctl."net.core.wmem_max"` changed from a string to an integer because of the addition of a custom merge option (taking the highest value defined to avoid conflicts between 2 services trying to set that value), just as `boot.kernel.sysctl."net.core.rmem_max"` since 22.11.
656:- `drbd` out-of-tree Linux kernel driver has been added in version 9.2.7. With it the DRBD 9.x features can be used instead of the 8.x features provided by the 8.4.11 in-tree driver.
672:- `lxd` has been upgraded to v5.21.x, an LTS release. The LTS release is now the only supported LXD release. Users are encouraged to [migrate to Incus](https://linuxcontainers.org/incus/docs/main/howto/server_migrate_lxd/) for better support on NixOS.
694:- `qt6.qtmultimedia` has changed its default backend to `QT_MEDIA_BACKEND=ffmpeg` (previously `gstreamer` on Linux or `darwin` on macOS).
783:- The kernel Yama LSM is now enabled by default, which prevents ptracing
786:  child). Or you can set `boot.kernel.sysctl."kernel.yama.ptrace_scope"` to 0.
```

```
rg -i "(linux|kernel)" nixos/doc/manual/release-notes/rl-2311.section.md
5:NixOS is a Linux distribution, whose set of packages can also be used on other Linux systems and macOS.
221:- `clang-ocl`, `clr`, `composable_kernel`, `hipblas`, `hipcc`, `hip-common`, `hipcub`,
301:- `util-linux` is now supported on Darwin and is no longer an alias to
302:  `unixtools`. Use the `unixtools.util-linux` package for access to the Apple
621:- `linuxPackages_testing_bcachefs` is now soft-deprecated by
622:  `linuxPackages_testing`.
623:  - Please consider changing your NixOS configuration's `boot.kernelPackages`
624:    to `linuxPackages_testing` until a stable kernel with bcachefs support is
697:  following the default in Linux Mint). If you still want Adwaita used, you may
875:- `zfs` was updated from v2.1.x to v2.2.0, [enabling newer kernel support and
995:- [Jool](https://nicmx.github.io/Jool/en/index.html), a kernelspace NAT64 and
1041:  [opensm](https://github.com/linux-rdma/opensm) systemd-template service,
1042:  instantiated on card guids. The module also adds kernel modules and cli
1307:- The Linux kernel module `msr` (refer to
1308:  [`msr(4)`](https://man7.org/linux/man-pages/man4/msr.4.html)), which provides
```

```
rg -i "(linux|kernel)" nixos/doc/manual/release-notes/rl-2305.section.md
5:NixOS is a Linux distribution, whose set of packages can also be used on other Linux systems and macOS.
17:- The default [Linux Kernel](https://kernel.org/) was updated from version 5.15 to 6.1, see [Kernelnewbies](https://kernelnewbies.org/Linux_6.1) for what has changed. All Kernels currently shown on [kernel.org](https://kernel.org/) are available.
41:- [Cinnamon](https://github.com/linuxmint/Cinnamon) has been updated to version 5.6, see [the pull request](https://github.com/NixOS/nixpkgs/pull/201328#issue-1449910204) for what was changed.
76:- [Deepin Desktop Environment](https://github.com/linuxdeepin/dde), an elegant, easy to use and reliable desktop environment. Available as [services.xserver.desktopManager.deepin](options.html#opt-services.xserver.desktopManager.deepin).
114:- [keyd](https://github.com/rvaiya/keyd), a key remapping daemon for Linux. Available as [services.keyd](#opt-services.keyd.enable).
166:- [v2rayA](https://v2raya.org), a Linux web GUI client of Project V which supports V2Ray, Xray, SS, SSR, Trojan and Pingtunnel. Available as [services.v2raya](options.html#opt-services.v2raya.enable).
339:- conntrack helper autodetection has been removed from kernels 6.0 and up upstream, and an assertion was added to ensure things don't silently stop working. Migrate your configuration to assign helpers explicitly or use an older LTS kernel branch as a temporary workaround.
347:  - Adjust `console.font` according to preference, though the kernel will generally choose a reasonably sized font
433:- To reduce closure size in `nixos/modules/installer/netboot/netboot-minimal.nix` profile disabled load linux firmwares, pre-installing the complete stdenv and `networking.wireless` service.
437:- NixOS installer ISOs can now be built for `powerpc64le-linux`; see `nixos/modules/installer/sd-card/sd-image-powerpc64le.nix` and [PR 192672](https://github.com/NixOS/nixpkgs/pull/192672).  Hydra does not support this platform, so you must build the binaries yourself.
592:- Nixpkgs now uses [IEEE-standard floating point arithmetic](https://github.com/NixOS/nixpkgs/pull/170215) on `powerpc64le-linux`.
```

</p>
</details>

**Comment by Ma27** [comment]
OK I'm a little surprised that 4.14 wasn't documented. However, for an LTS kernel I'd consider a not warranted.

**Comment by fabianhjr** [comment]
Copied over the same relaese notes from the 22.11 drop of 4.9 LTS kernel and put it near other kernel entries of the 24.11 release notes.

**Review by Ma27 — APPROVED** [review]

**Pull request merged**